### PR TITLE
Update np_trainer.py

### PR DESCRIPTION
**Pull request by CongWeilin**
*Issue #, if available:*

when using precision_recall as evaluation metric, we need to set `return_proba=True` because scikit-learn's precision_recall_curve takes probability as input.

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: python/graphstorm/trainer/np_trainer.py [hunk 1]
```diff
@@ -167,9 +168,10 @@ def fit(self, train_loader, num_epochs,
                 if self.evaluator is not None and \
                     self.evaluator.do_eval(total_steps, epoch_end=False) and \
                     val_loader is not None:
+
                     val_score = self.eval(model.module if is_distributed() else model,
```
Comment (classicsong) [inline python/graphstorm/trainer/np_trainer.py:172]: Can we modify the eval() function instead of the fit() function?
And we should add an warning here that we force the return_proba = True.

File: python/graphstorm/trainer/np_trainer.py [hunk 2]
```diff
@@ -261,6 +261,12 @@ def eval(self, model, val_loader, test_loader, use_mini_batch_infer, total_steps
         """
         teval = time.time()
         sys_tracker.check('before prediction')
+
+
+        if 'precision_recall' in self.evaluator.metric and return_proba==False:
```
Comment (classicsong) [inline python/graphstorm/trainer/np_trainer.py:266]: compute_roc_auc, compute_precision_recall_auc

File: python/graphstorm/trainer/np_trainer.py [hunk 3]
```diff
@@ -261,6 +261,17 @@ def eval(self, model, val_loader, test_loader, use_mini_batch_infer, total_steps
         """
         teval = time.time()
         sys_tracker.check('before prediction')
+
+        metric = set(self.evaluator.metric)
+        need_proba = metric.intersection({'roc_auc', 'per_class_roc_auc', 'precision_recall'})
+        need_label_pred = metric.intersection({'accuracy', 'f1_score', 'per_class_f1_score'})
+        assert len(need_proba) == 0 or len(need_label_pred) == 0, \
+            print(f"{need_proba} requires return_proba==True, \
```
Comment (classicsong) [inline python/graphstorm/trainer/np_trainer.py:269]:
```suggestion
            f"{need_proba} requires return_proba==True, \
```

File: python/graphstorm/trainer/np_trainer.py [hunk 4]
```diff
@@ -261,6 +261,17 @@ def eval(self, model, val_loader, test_loader, use_mini_batch_infer, total_steps
...
+                         but {need_label_pred} requires return_proba==False.")
```
Comment (classicsong) [inline python/graphstorm/trainer/np_trainer.py:270]:
```suggestion
                         but {need_label_pred} requires return_proba==False."
```

File: python/graphstorm/trainer/np_trainer.py [hunk 5]
```diff
@@ -261,6 +261,17 @@ def eval(self, model, val_loader, test_loader, use_mini_batch_infer, total_steps
...
+        if len(need_proba) > 0 and return_proba==False:
+            return_proba = True
+            logging.info(f"{need_proba} requires return_proba==True.")
```
Comment (classicsong) [inline python/graphstorm/trainer/np_trainer.py:273]:
```suggestion
            logging.warning(f"{need_proba} requires return_proba==True. Set return_proba to True")
```

**Review by classicsong — COMMENTED** [review]
Can you also change ep_trainer.py?

**Review by classicsong — APPROVED** [review]
LGTM
